src/flow_checks/RareDestination.cpp

```diff
@@ -24,16 +24,89 @@
 
 /* ***************************************************** */
 
+u_int32_t RareDestination::getDestinationHash(Flow *f) {
+
+  u_int32_t hash = 0;
+  Host *dest = f->get_srv_host();
+
+  if (f->isLocalToLocal()) {
+
+    char buf[64];
+
+    if (!dest->isMulticastHost() && dest->isDHCPHost()) {
+      char *mac = dest->getMac()->print(buf,sizeof(buf));
+      hash = Utils::hashString(mac);
+    }
+    else if (dest->isIPv6() || dest->isIPv4()) {
+      char *ip = dest->get_ip()->print(buf,sizeof(buf));
+      hash = Utils::hashString(ip);
```

Comment:
This is not a string, please use a better hash function (see as example HostHash::get())